cmd/api/src/database/agi.go

```diff
@@ -86,47 +86,74 @@ func (s *BloodhoundDB) DeleteAssetGroup(ctx context.Context, assetGroup model.As
 	})
 }
 
+func (s *BloodhoundDB) getAssetGroupMemberCount(ctx context.Context, assetGroupID int32) (int, error) {
+    var (
+        latestCollection model.AssetGroupCollection
+        result           = s.preload(model.AssetGroupCollectionAssociations()).
+                             WithContext(ctx).
+                             Where("asset_group_id = ?", assetGroupID).
+                             Order("created_at DESC").
+                             First(&latestCollection)
+    )
+
+    if result.Error != nil {
+        if errors.Is(result.Error, gorm.ErrRecordNotFound) {
+            return 0, nil
+        }
+        return 0, fmt.Errorf("error getting latest collection for asset group ID %d: %w", assetGroupID, result.Error)
+    }
```

Comment:
This could be unnested as:

```go
if errors.Is(result.Error, gorm.ErrRecordNotFound) {
    return 0, nil
} else if result.Error != nil {
    return 0, fmt.Errorf("error getting latest collection for asset group ID %d: %w", assetGroupID, result.Error)
}
```

Reply:
See @zinic's comment, as I didn't see it before adding this. We want to use the stdlib version of `errors` here (may need to change other usages in this file though, so if it gets crazy feel free to ask for assistance), and this is roughly what @zinic was talking about with the unnecessary nil check.

Reply:
Given that `stdErrors.Is` is nil-safe, do I still need to add `result.Error != nil`, having made the switch?

Reply:
You probably answered your own question judging by the edit, but just for anyone else wondering: 

Yes, you want to do your `errors.Is` branches first to check for specific errors you want to add special handling for. After that, you should always do a nil check to see if you need to handle general errors you're not specifically interested in.